lib/Backend/Vault/Kv/Internal.hs

```diff
@@ -0,0 +1,330 @@
+-- >      "custom_metadata":
+-- >      { "owner": "jdoe",
+-- >        "mission_critical": "false"
+-- >      },
+-- >      "deletion_time": "",
+-- >      "destroyed": false,
+-- >      "version": 2
+-- >    }
+-- >  }
+data ReadSecret =
+  ReadSecret
+  { _secret :: HS.HashMap T.Text T.Text
+  , _rsCustomMetadata :: HS.HashMap T.Text T.Text
+  , _createdTime :: T.Text
+  , _deletionTime :: T.Text
+  , _destroyed :: Bool
+  , _version :: Int
+  }
+  deriving (Show)
+makeLenses ''ReadSecret
+makeLensesWith abbreviatedFields ''ReadSecret
+
+
+-- |
+-- A type defining the request to the Vault servet at endpoint
+-- '/v1/<MOUNT>/data/<PATH>'.
+-- For example:
+--
+-- >  { "options":
+-- >  , { "cas": 0
+-- >    }
+-- >  , "data":
+-- >    { "foo": "bar"
+-- >    , "zip": "zap"
+-- >    }
+-- >  }
+data PostSecret =
+  PostSecret
+  { _cas :: Maybe Int
+  , _psDdata :: HS.HashMap T.Text T.Text
+  }
+  deriving (Show)
+type PatchSecret = PostSecret
+makeLenses ''PostSecret
+makeLensesWith abbreviatedFields ''PostSecret
+
+
+-- |
+-- A type defining the request to the Vault servet at endpoint
+-- '/v1/<MOUNT>/metadata/<PATH>'.
+-- For example:
+--
+-- >  { "max_versions": 5
+-- >  , "cas_required": false
+-- >  , "delete_version_after": "40m"
+-- >  , "custom_metadata":
+-- >    { "example": ":)"
+-- >    }
+-- >  }
+data UpdateMetadata =
+  UpdateMetadata
+  { _maxVersions :: Maybe Int
+  , _casRequired :: Maybe Bool
+  , _deleteVersionAfter :: Maybe T.Text
+  , _umCustomMetadata :: HS.HashMap T.Text T.Text
+  }
+  deriving (Show)
+makeLenses ''UpdateMetadata
+makeLensesWith abbreviatedFields ''UpdateMetadata
+
+-- Overloaded Lens accessors
+
+instance {-# OVERLAPPABLE #-} a ~ b => HasDdata (KvResponse a) b where
+  ddata = kdata
+
+-- JSON serialization/deserialization, logically some ADTs need to only be deserialized,
+-- others serialized, but never both.
+
+instance FromJSON ListSecrets where
+  parseJSON = withObject "ListSecrets" $ \o ->
+    ListSecrets <$> o .: "keys"
+
+instance FromJSON ReadSecret where
+  parseJSON = withObject "ReadSecret" $ \o -> do
+    metadata <- o .: "metadata"
+    ReadSecret
+      <$> o .: "data"
+      <*> (metadata .:! "custom_metadata" <|> error "Cannot parse 'custom_metadata' field") .!= HS.fromList []
+      <*> metadata .: "created_time"
+      <*> metadata .: "deletion_time"
+      <*> metadata .: "destroyed"
+      <*> metadata .: "version"
+
+instance FromJSON a => FromJSON (KvResponse a) where
+  parseJSON = withObject "KvResponse" $ \o ->
+    KvResponse
+    <$> o .: "request_id"
+    <*> o .: "lease_id"
+    <*> o .: "renewable"
+    <*> o .: "lease_duration"
+    <*> o .: "data"
+
+removeNull :: [AT.Pair] -> [AT.Pair]
+removeNull = filter \case 
+  (_, Null) -> False 
+  _         -> True
+
+instance ToJSON PostSecret where
+  toJSON postSecret =
+    object [ "options" .= object (removeNull [ "cas" .= (postSecret ^. cas)  ])
+             , "data" .= (postSecret ^. ddata)
+             ]
```

Comment:
Because we removed the `A.` qualified, the `[` and `,` aren't aligned anymore.

The [styleguide](https://github.com/serokell/style/blob/master/haskell.md#list-declarations) recommends multi-line lists should go in a separate line - to avoid alignment issues during refactoring.

```suggestion
    object
      [ "options" .= object (removeNull [ "cas" .= (postSecret ^. cas)  ])
      , "data" .= (postSecret ^. ddata)
      ]
```

Reply:
Done